lib/services/history/sources/remote_history.ts

```diff
@@ -0,0 +1,78 @@
+import { SNApiService } from "@Lib/services/api/api_service";
+import { SNProtocolService } from "@Lib/services/protocol_service";
+import { ItemHistory } from "../item_history";
+import { HttpResponse } from "@Lib/services/api/http_service";
+import { RawPayload, CreateMaxPayloadFromAnyObject, CreateSourcedPayloadFromObject } from "@Lib/protocol/payloads/generator";
+import { ItemHistorySource, ItemHistoryEntry } from "../item_history_entry";
+import { PayloadSource } from "@Lib/protocol/payloads";
+
+type RevisionResponse = {
+  error?: any
+  revision?: RawPayload
+  status?: number
+}
+
+type RevisionsResponse = {
+  error?: any
+  revisions?: RawPayload[]
+  status?: number
+}
+
+export class RemoteHistory {
+  private apiService: SNApiService
+  private protocolService: SNProtocolService
+
+  constructor(apiService: SNApiService, protocolService: SNProtocolService) {
+    this.apiService = apiService;
+    this.protocolService = protocolService;
+  }
+
+  /**
+   * Iterates over the response and creates a payload from each entry.
+   */
+  private responseToPayloadArray(response: RevisionsResponse) {
+    const revisions = response.revisions!;
+    const revisionEntries = revisions.map((revision) => {
+      return {
+        payload: CreateMaxPayloadFromAnyObject(revision)
```

Comment:
Wouldn't the payload.uuid at this point be the uuid of the revision and not the item? We don't seem to be replacing the uuid like we do in `fetchItemRevision`.

If this is the case, you should:

1. Create a test that catches this issue.
2. Create a reusable function `rawRevisionToItemPayload` that cleans/restructures revision responses as needed and creates payloads from them.

Reply:
> Wouldn't the payload.uuid at this point be the uuid of the revision and not the item? We don't seem to be replacing the uuid like we do in fetchItemRevision.

Yes, but we don't need to replace it in this case, the revision doesn't have a `content` field and thus will not be decryptable.

Reply:
Ok, then we need to create a type that describes exactly what the response is, to avoid this kind of confusion. It should be very strictly defined what a `revisions` element looks like.

Reply:
This should be the types:

```javascript
type RawRevision = RawPayload & {
  item_id: string
}
```

```javascript
type RawRevisionsListItem = {
  uuid: string
  content_type: ContentType
  created_at: string
  updated_at: string
}
```

`getItemRevisions` should return an array of `RawRevisionsListItem`.

But since the body fields are combined with the top level response object, I will have to parse the response and iterate the object keys.